CMakeLists.txt

```diff
@@ -282,6 +282,11 @@ endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
   # u8path() function is deprecated but there is no sensible alternative and it might even get un-deprecated.
   add_definitions(-D_SILENCE_CXX20_U8PATH_DEPRECATION_WARNING)
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug" AND ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.25")

```

Comment:
I don't think `CMAKE_BUILD_TYPE` works properly in multi-config  IDE such as Visual Studio, so you need to use a `$<CONFIG:Debug>` generator expression (like in the example in CMake documentation).

Reply:
I changed it to

```CMake
  if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.25")
    # This enables Edit & Continue support, see https://learn.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio#edit-and-continue-for-cmake-projects
    set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "$<$<CONFIG:Debug>:EditAndContinue>") # Sets /ZI compiler option, see https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_DEBUG_INFORMATION_FORMAT.html
    add_link_options("$<$<CONFIG:Debug>:/INCREMENTAL>")
  endif()
```

and removed the `CMAKE_BUILD_TYPE` check.
Do I need an additional protection for msvc? (as in your example above)

Thanks for the help glebm. ❤️

Reply:
>  Do I need an additional protection for msvc? (as in your example above)

Not according to cmake documentation:

> The value is ignored on compilers not targeting the MSVC ABI

However, it does require setting CMP0141 to NEW

> This variable has effect only when policy [CMP0141](https://cmake.org/cmake/help/latest/policy/CMP0141.html#policy:CMP0141) is set to NEW

Reply:
I added the CMP0141 policy in the beginning of the CMake file, where we also set other policies to NEW.

Another question:
We also use `CMAKE_BUILD_TYPE STREQUAL "Debug"` in

https://github.com/diasurgical/devilutionX/blob/5ed7ca98089ca95589e3f379e7755d0cdc241791/CMakeLists.txt#L197-L202

should this also be changed (in another PR)?

Reply:
This is already inside `if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")` so it should be fine.

Reply:
>  We also use CMAKE_BUILD_TYPE STREQUAL "Debug"

I'm not sure how to handle that one or whether it even works as expected.
It's a bit tricky because `BUILD_TESTING` is used at configure-time to enable the test targets.

Reply:
I tried it, but it didn't work for me. 😢 
But that is not directly related to this PR.

I did some reconfiguring and rebuilding with this PR and it looks like it works. Do you think the PR is ok this way? 🙂

Reply:
I think it's fine but while I know a bit about CMake I don't use Visual Studio at all, so let's get @StephenCWills or someone else who uses VS to also have a look.